src/Env/EnvFactory.php

```diff
@@ -27,8 +29,8 @@ class EnvFactory
      *
      * @return Env
      */
-    public static function createEnv(EnvEnum $type)
+    public static function createEnv(EnvEnum $type, AppVendor $appVendor, Dependency ...$dependencies)
```

Comment:
BTW, not sure I like the varargs in this factory method. When you'll introduce a new argument, this one'll have to kept being the last one. 
If the newly introduced var is an array too, what would be the legitimate reason to have one variadic but another not ?

IMHO variadic functions/methods must not be overused and does only make sense to very specific cases.